crates/oxc_regular_expression/src/display.rs

```diff
@@ -0,0 +1,541 @@
+        // we lose the flags ordering
+        ("/abcd/igv", Some("/abcd/giv"), true),
+        // we lose the flags ordering
+        (r"/\d/ug", Some(r"/\d/gu"), true),
+        // NOTE: we capitalize all hex values.
+        (r"/\n\cM\0\x41\u{1f600}\./u", Some(r"/\n\cM\0\x41\u{1F600}\./u"), true),
+        // TODO: tests below do **NOT** parse with unicode mode on
+        (r"/c]/", None, false),
+        // Octal tests from: <https://github.com/tc39/test262/blob/d62fa93c8f9ce5e687c0bbaa5d2b59670ab2ff60/test/annexB/language/literals/regexp/legacy-octal-escape.js>
+        (r"/\1/", None, false),
+        (r"/\2/", None, false),
+        (r"/\3/", None, false),
+        (r"/\4/", None, false),
+        (r"/\5/", None, false),
+        (r"/\6/", None, false),
+        (r"/\7/", None, false),
+        // NOTE: we remove leading zeroes
+        (r"/\00/", Some(r"/\0/"), false),
+        // NOTE: we remove leading zeroes
+        (r"/\07/", Some(r"/\7/"), false),
+        (r"/\40/", None, false),
+        (r"/\47/", None, false),
+        (r"/\70/", None, false),
+        (r"/\77/", None, false),
+        // NOTE: we remove leading zeroes
+        (r"/\000/", Some(r"/\0/"), false),
+        // NOTE: we remove leading zeroes
+        (r"/\007/", Some(r"/\7/"), false),
+        // NOTE: we remove leading zeroes
+        (r"/\070/", Some(r"/\70/"), false),
+        (r"/\300/", None, false),
+        (r"/\307/", None, false),
+        (r"/\370/", None, false),
+        (r"/\377/", None, false),
+        (r"/(.)\1/", None, false),
+        // Identity escape from: <https://github.com/tc39/test262/blob/d62fa93c8f9ce5e687c0bbaa5d2b59670ab2ff60/test/annexB/language/literals/regexp/identity-escape.js>
+        (r"/\C/", None, false),
+        (r"/O\PQ/", None, false),
+        (r"/\8/", None, false),
+        (r"/7\89/", None, false),
+        (r"/\9/", None, false),
+        (r"/8\90/", None, false),
+        // TODO: fails to parse
+        // (r"(.)(.)(.)(.)(.)(.)(.)(.)\8\8", None, false),
+        // Class escape from: <https://github.com/tc39/test262/blob/d62fa93c8f9ce5e687c0bbaa5d2b59670ab2ff60/test/annexB/language/literals/regexp/class-escape.js>
+        (r"/\c0/", None, false),
+        (r"/[\c0]/", None, false),
+        (r"/\c1/", None, false),
+        (r"/[\c10]+/", None, false),
+        (r"/\c8/", None, false),
+        (r"/[\c8]/", None, false),
+        (r"/[\c80]+/", None, false),
+        (r"/\c_/", None, false),
+    ];
+
+    fn test_display(allocator: &Allocator, (source, expect, unicode): Case) {
+        use crate::{Parser, ParserOptions};
+        let expect = expect.unwrap_or(source);
+        let parsed_utf16 =
+            Parser::new(allocator, source, ParserOptions::default()).parse().unwrap();
+        assert_eq!(expect, parsed_utf16.to_string());
+        if unicode {
+            let parsed_unicode =
+                Parser::new(allocator, source, ParserOptions::default().with_unicode_mode())
```

Comment:
If you are using `(Literal)Parser`, flag options are useless. They are already handled. 😉 

https://github.com/oxc-project/oxc/blob/afb038e93e85986e5fc229d8db4a5d13154a5184/crates/oxc_regular_expression/src/literal_parser.rs#L27

Reply:
Please correct me if I'm wrong, I thought it would use the `Parser::options` field, It would only override unicode flags if they are passed in the literal. I want to test the same literal with Unicode and utf-16 mode to ensure we are handling surrogated characters and their u32 code points the same way.

Reply:
> It would only override unicode flags if they are passed in the literal

Yes, and no.

- `Parser("/abc/u", Options::default())`: true
- `Parser("/abc/u", Options { unicode_mode: false })`: true(!?)
- `Parser("/abc/", Options { unicode_mode: true })`: true

> I want to test the same literal with Unicode and utf-16 mode

For that purpose, I recommend to use `PatternParser`.

Reply:
I agree using `PatternParser` is probably better for testing this. Maybe `Parser` shouldn't take these options if we intend to ignore them.
I think this should be either true or panic/raise diagnostic error

> Parser("/abc/u", Options { unicode_mode: false })